apollo-mockserver-junit5/src/test/java/com/ctrip/framework/apollo/mockserver/ExtensionLifecycleTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2023 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.mockserver;
+
+import com.ctrip.framework.apollo.Config;
+import com.ctrip.framework.apollo.ConfigService;
+import com.ctrip.framework.apollo.core.ConfigConsts;
+import com.ctrip.framework.apollo.model.ConfigChangeEvent;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+
+@ExtendWith(MockApolloExtension.class)
+public class ExtensionLifecycleTest {
+
+  private ApolloTestingServer target;
+
+  @AfterEach
+  public void after() {
+    Assertions.assertTrue(target.isClosed());
+  }
+
+  @Test
+  public void testParameterInjection(ApolloTestingServer server) throws Exception {
+    Assertions.assertTrue(server.isStarted());
+    target = server;
+    Config applicationConfig = ConfigService.getAppConfig();
+
+    CompletableFuture<ConfigChangeEvent> future = new CompletableFuture<>();
+    applicationConfig.addChangeListener(future::complete);
+
+    assertEquals("value1", applicationConfig.getProperty("key1", null));
+    assertEquals("value2", applicationConfig.getProperty("key2", null));
+
+    server.addOrModifyProperty(ConfigConsts.NAMESPACE_APPLICATION, "key2", "newValue2");
```

Comment:
Shall we clear the state in the `after` method?

Reply:
What state? You means local cache?

Reply:
the overridden properties, or it may affect other tests

Reply:
Don't worry about that:
- ApolloTestingServer  will be closed(close -> clear -> resetOverriddenProperties) automatically by MockApolloExtension.
- Overridden properties will be double-cleared on testing server start.

Reply:
Since there is no other test in the apollo-mockserver-junit5, there is no impact.
What I did was to merge the codes into apollo-mockserver, which I found it failed other tests.

![image](https://user-images.githubusercontent.com/837658/236589330-bbeee5d5-3a18-4a85-ba47-7cf36b73b459.png)

Reply:
I confirmed that ApolloTestingServer.close() and resetOverriddenProperties() were invoked. BUT, it seems that not work on default namespace.

Reply:
You may update existing ApolloMockServerApiTest.testUpdateProperties(), use getSysConfig() instead of 'anotherNamespace', then some other tests will fail for that.

Reply:
It's funny, I'm reading the code to find out why.

Reply:
I got it! The root cause is sharing of `Config` object, and we are not able to clear cache explicitly. If you add a Thread.sleep(5000) before testGetProperty, it will pass because delayed change event will be received anyway.